src/solvers/refinement/README.md

```diff
@@ -6,59 +6,101 @@
 
 \section string_solver_interface String solver interface
 
+The basic role of the solver is to answer whether the set of equations given
+is satisfiable. One example usage, is to determine whether an assertion in a
+program can be violated.
+For instance, CBMC and JBMC, convert a input program and property to check
+about this program to a set of equations. These equations are fed to a solver,
+which is one of the last step in CBMC and JBMC, as it tells us whether the
+property holds or can fail.
+
+The secondary role of the solver is to provide a satisfying assignment of
+the variables of the equations, this can for instance be used to construct
+a trace.
+
+The string solver is particularly aimed at string logic, but since it inherits
```

Comment:
I think the first three paragraphs here can be made more concise. Someone looking at this code can be expected to know that the directory `solvers`  is about satisfiability solvers. It is much more important to explain how the classes in `refinement` add features to the SMT backend and how they build on and interact with other classes in `solvers`.

Reply:
> Someone looking at this code can be expected to know that the directory solvers is about satisfiability solvers

The idea was to give some indication to someone not knowing what is in this part of the code, so I think it's good to have some high level explanation somewhere. 
For the explanation about the interaction between the different solver, we touch a bit the subject in the documentation of `dec_solve` but this is quite minimal as the interactions are quite limited.
I think we would need a more general description of the solver architecture in cbmc.